### ISSUE TITLE
`react-select` not working as expected with JAWS

### ISSUE THREAD

**Issue by esizer** [GCTC-NTGC/gc-digital-talent#3386]
## Summary

Currently there are some issues when interacting with `react-select` using JAWS. We were provided some feedback from Fable and it will be outlined in this issue.

### Field Appears Blank After Selection

"When I engage the drop-down list for the category filter, I am able to find the option "IT-01" with no problems. However, when I then leave forms mode after selecting it, the field appears blank when I use the read line command ("JAWS key + I") to double-check the filter had been properly applied. This makes me wonder if the form has properly updated as I receive no information about the form updating."

### Improper Focus Management

"When I entered "national capital" into the forms field, I was presented with a result to select below the field. **I selected the option with "Spacebar"**, and the form immediately moved my focus to a random location of the form."

"When I entered "National capital" into the region field, I was presented with an option to select and promptly selected it with "space bar". My focus was then randomly move to the top of the page, which prompted me to try to re-find the region field. I then noticed that section of the field had not presented any information as to whether or not the filter had been applied, let alone what region. I checked the filters just above the search results, and all the information I was given was simply "'same page link' Classification Filters (1)" (JAWS narrated that this link to remove the filter was a same page link. I am uncertain what filters, if any, have been applied at this point in the form."

 Relevant WCAG criteria:  

-  3.2.1 On Focus 
- 3.2.2 On Input 

### State Not Being Carried Over to Request Form

"Upon reviewing the information for the form, I noticed there is a section below it that lists the filters applied. For some reason, the location "National capital" was not applied. I think that the field for locating such a location in the form must be either broken or not properly registering the text JAWS is in-putting as actual text visually."

## Notes

Ref: https://talent-cloud.slack.com/archives/CFCE2FSR4/p1658173082884179

There was a discussion where @patcon identified that it may be an issue with instruction on how to use the field. `react-select` replaces a traditional `select` element with an `input[type=text]` that has a dropdown. By the sounds of it, the user was attempting to interact with it like it was a traditional `select` input. 

We may need to provide better instructions and associate them with the input using `aria-describedby`.

**Comment by patcon** [comment]
To confirm, this test being done on the prod page, right?
https://talent.canada.ca/en/talent/search

(a snapshotted version of storybook might be helpful for testing this moment in time, in case we push changes to the prod page that materially changes the page as we troubleshoot
https://main--61e099a1bb1465003a73d3ce.chromatic.com/?path=/story/search-page--search-page)

**Comment by patcon** [comment]
I'm just seeing Andrew as being on "JAWS with Chrome". Does anyone know if Fable testers are guaranteed to always be on the latest versions of software, or is JAWS version something we should at least be aware of?

**Comment by patcon** [comment]
While writing the tests, I noticed that this is already shown when using the Select

```
<span id="aria-context">
    Use Up and Down to choose options, press Enter to select the currently focused option, press Escape to exit the menu, press Tab to select the option and exit the menu.
</span>
```

but maybe this not being read on JAWS is why this one person was trying to use "space" and everyone else was giving good reviews. If this is the case, this might be something to bring upstream

**Comment by esizer** [comment]
> While writing the tests, I noticed that this is already shown when using the Select

Does the input include an `aria-describedby="aria-context"`?

**Comment by patcon** [comment]
Sorry @esizer -- added more context above :)

**Comment by patcon** [comment]
If it ends up being this message that needs improvement, I think this is where we hook in to change it:

- the prop: https://react-select.com/props#:~:text=%3E-,ariaLiveMessages,-Customise%20the%20messages
- where the prop is generated in react-select codebase: https://github.com/JedWatson/react-select/blob/4b8468636bcfadf0cfe45f9a7a6c1db5dca08d9a/packages/react-select/src/components/LiveRegion.tsx#L161-L189

**Comment by gobyrne** [comment]
I don't think we can assume Fable users are using the most up-to-date version of JAWS or similar. They are probably, on average, more up to date than other screen-reader user though.

**Comment by patcon** [comment]
## Concrete Proposal

To be concrete, I'm advocating that we should change the instruction message from:

> Use Up and Down to choose options, press Enter to select the currently focused option, press Escape to exit the menu, press Tab to select the option and exit the menu.

to this

> Press Enter to select the currently focused option, press Tab to select the option and exit the menu, press Escape to exit the menu, use Up and Down to choose options.

Basically, lead with the most non-intuitive option, and say the "enter" bit asap before someone is tempted to skip the instructions.

We can both:
- fix this locally, using the links and hooks mentioned earlier
- create ticket and advocate for this change upstream 🎉 

## Validating with Affected Users

Validating this with affected users is the ideal practice, but now that I'm reading the existing instructions, the improved order seems like a no-brainer. Saying "up/down" should not be first, as no one is unaware of that :)

**Comment by esizer** [comment]
> Press Enter to select the currently focused option, press Tab to select the option and exit the menu, press Escape to exit the menu, use Up and Down to choose options.

Wait, our select component uses `tab` to select items? This seems counterintuitive to how keyboard navigation works. `tab` is used for navigation and not selection. If a user presses `tab` in `select` or dropdown it typically navigates **_without_** selection 🤔 

Is it possible to remove the ability to select items with `tab` and just have it close the menu and navigate to the next focusable item?

Also, it would be nice to have `space` select an item as well which is another common command used for interaction like with `input[type=checkbox]`

**Comment by mnigh** [comment]
Hoping that the issue is solved in #5507.
